src/relayer/Relayer.t.sol

```diff
@@ -318,4 +350,106 @@ contract RelayerTest is DSTest {
+        // For this test we will simulate an external actor that must be able to successfully update
+        // the Relayer via multiple executeWithRevert() calls. We will do this by providing multiple
+        // mocked chainlink values and making sure that each value is properly validated and used.
+        int256 firstValue = 1e18;
+        int256 secondValue = 2e18;
+        uint256 timeUpdateWindow = 100;
+
+        // Test setup, create the mockChainlinkAggregator that will provide data for the Oracle
+        MockProvider mockChainlinkAggregator = new MockProvider();
+        mockChainlinkAggregator.givenQueryReturnResponse(
+            abi.encodeWithSelector(
+                IChainlinkAggregatorV3Interface.decimals.selector
+            ),
+            MockProvider.ReturnData({success: true, data: abi.encode(18)}),
+            false
+        );
+
+        // Deploy the chainlink Oracle with the mocked chainlink datafeed contract
+        ChainlinkValueProvider chainlinkVP = new ChainlinkValueProvider(
+            timeUpdateWindow,
+            address(mockChainlinkAggregator)
+        );
+
+        // Deploy the Relayer
+        Relayer testRelayer = new Relayer(
+            address(collybus),
+            relayerType,
+            address(chainlinkVP),
+            _mockTokenId,
+            _mockMinThreshold
+        );
+
+        // Whitelist the relayer in the oracle so it can trigger Oracle.update()
+        chainlinkVP.allowCaller(chainlinkVP.ANY_SIG(), address(testRelayer));
+
+        // Simulate a small time offset as the Oracle.lastTimestamp will start at 0 and we won't be able to update values
+        uint256 timeStart = timeUpdateWindow + 1;
+
+        // Set the Chainlink mock to return the first value
+        setChainlinkMockReturnedValue(mockChainlinkAggregator, firstValue);
+
+        // Forward time to the start
+        hevm.warp(timeStart);
+
+        // Run the first execute with revert, Oracle.currentValue and Oracle.nextValue will be equal to firstValue
+        testRelayer.executeWithRevert();
+
+        // Set the next value for the chainlink mock datafeed
+        setChainlinkMockReturnedValue(mockChainlinkAggregator, secondValue);
+
+        // Forward time to the next window
+        hevm.warp(timeStart + timeUpdateWindow);
+
+        // Call should not revert
+        testRelayer.executeWithRevert();
+
+        // Verify that the oracle was properly updated
+        // The nextValue should now be equal to `secondValue` because of the update
+        assertTrue(
+            chainlinkVP.nextValue() == secondValue,
+            "Invalid Oracle nextValue"
+        );
+
+        // The current oracle value should still be the first value because fon the previous executeWithRevert()
+        // currentValue and nextValue where initialized to `firstValue`
+        (int256 value, bool isValid) = chainlinkVP.value();
+        assertTrue(isValid && value == firstValue, "Invalid Oracle value");
```

Comment:
In testing it's better to split checks into multiple asserts, since we want to check things independently and we don't care about gas costs.